canonical-web-and-design/maas-ui
# feat(ui): add ability to download generated certificates

## Done

- Added ability to download generated certificates from the UI

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and click Add KVM
- Enter details and keep the "Generate new certificate" radio selected
- Click the "Download certificate" button and save the file
- Check that the file contents are correct

## Fixes

Fixes canonical-web-and-design/app-squad#248
